src/main/java/frc/robot/subsystems/Swerve.java

```diff
@@ -0,0 +1,324 @@
+                new SparkFlexSwerve(RobotMap.SWERVE_BACK_LEFT_DRIVE,true,DCMotor.getNeoVortex(1)),
+                new SparkFlexSwerve(RobotMap.SWERVE_BACK_LEFT_STEER,false,DCMotor.getNeoVortex(1)),
+                conversionFactor,
+                new CANCoderSwerve(RobotMap.SWERVE_BACK_LEFT_ENCODER),
+                258.75,
+                -LENGTH / 2,
+                WIDTH / 2,
+                steerPidf,
+                drivePidf,
+                characteristics,
+                false,
+                false,
+                false,
+                "BackLeft",
+                false
+        );
+        SwerveModuleConfiguration backRight = new SwerveModuleConfiguration(
+                new SparkFlexSwerve(RobotMap.SWERVE_BACK_RIGHT_DRIVE,true,DCMotor.getNeoVortex(1)),
+                new SparkFlexSwerve(RobotMap.SWERVE_BACK_RIGHT_STEER,false,DCMotor.getNeoVortex(1)),
+                conversionFactor,
+                new CANCoderSwerve(RobotMap.SWERVE_BACK_RIGHT_ENCODER),
+                177,
+                -LENGTH / 2,
+                -WIDTH / 2,
+                steerPidf,
+                drivePidf,
+                characteristics,
+                false,
+                false,
+                false,
+                "BackRight",
+                false
+        );
+        SwerveDriveConfiguration configuration = new SwerveDriveConfiguration(
+                new SwerveModuleConfiguration[] {
+                        frontLeft, frontRight, backLeft, backRight
+                },
+                new Pigeon2Swerve(RobotMap.SWERVE_PIGEON),
+                false,
+                characteristics
+        );
+        SwerveControllerConfiguration controllerConfiguration = new SwerveControllerConfiguration(
+                configuration,
+                new PIDFConfig(0.4, 0, 0.01),
+                0.5,
+                MAX_SPEED
+        );
+
+        SwerveDriveTelemetry.verbosity = SwerveDriveTelemetry.TelemetryVerbosity.HIGH;
+
+        swerveDrive = new SwerveDrive(configuration, controllerConfiguration, MAX_SPEED, Pose2d.kZero);
+        swerveDrive.setHeadingCorrection(false);
+        swerveDrive.setCosineCompensator(false);
+        swerveDrive.setAngularVelocityCompensation(false, false, 0);
+        swerveDrive.setModuleEncoderAutoSynchronize(false, 1);
+        swerveDrive.pushOffsetsToEncoders();
+
+        swerveDrive.resetOdometry(Pose2d.kZero);
+
+        mechanism = new Mechanism2d(50, 50);
+        moduleMechanisms = createMechanismDisplay(mechanism);
+        SmartDashboard.putData("SwerveMechanism", mechanism);
+
+        PathPlannerLogging.setLogActivePathCallback((poses)-> {
+            swerveDrive.field.getObject("trajectory").setPoses(poses);
+        });
+        limeLights = new LimeLights();
+        secondField = new Field2d();
+    }
+
+    public Command driveA(DoubleSupplier translationX, DoubleSupplier translationY, DoubleSupplier angularRotationX) {
+        return run(() -> {
```

Comment:
use `runEnd` and for the end phase use a `stop` method. You can implement it from the `stop` logic in `setSpeeds`